articles/mongodb-nodejs-custom-ranking/index.md

```diff
@@ -0,0 +1,204 @@
+    if (err) throw err;
+    console.log("Collection created!");
+    db.close();
+  });
+});
+```
+
+#### *Inserting documents to our table*
+
+We will insert some dummy data of 10 students into our collection for demonstrating the topic of the article.
+Create a file called *insert.js* and then add the following code.
+
+```javascript
+var MongoClient = require('mongodb').MongoClient;
+var url = "mongodb://localhost:27017/";
+
+MongoClient.connect(url, function(err, db) {
+  if (err) throw err;
+  var dbo = db.db("ourDB");
+  var myobj = [
+    { name: 'John', mark: '71'},
+    { name: 'Peter', mark: '44'},
+    { name: 'Amy', mark: '52'},
+    { name: 'Hannah', mark: '21'},
+    { name: 'Michael', mark: '21'},
+    { name: 'Sandy', mark: '51'},
+    { name: 'Betty', mark: '22'},
+    { name: 'Richard', mark: '11'},
+    { name: 'Susan', mark: '88'},
+    { name: 'Viola', mark: '71'}
+  ];
+  dbo.collection("ourTable").insertMany(myobj, function(err, res) {
+    if (err) throw err;
+    console.log("Number of documents inserted: " + res.insertedCount);
+    db.close();
+  });
+});
+```
+
+#### *Querying and giving ranks to our documents*
+
+This is where the core of this article is. We will query and sort the students using their marks in descending order. We will then populate the total marks to an array then perform conditional checks to check for ties and give ranks appropriately.
+
+We create a new file called *rank.js* then add the following code.
+
+```javascript
+var MongoClient = require('mongodb').MongoClient;
+var url = "mongodb://localhost:27017/";
+let rankArray = [];
+let dbo;
+
+MongoClient.connect(url, function(err, db) {
+    if (err) throw err;
+    dbo = db.db("ourDB");
+    loadStudents();
+    db.close();
+});    
+
+
+function loadStudents(){
+        var mysort = { mark: -1 };
+        dbo.collection("ourTable").find().sort(mysort).toArray(function(err, result) {
+          if (err) throw err;
+          for (let i = 0; i < result.length; i++) {
+              rankArray[i] = result[i]['mark'];
+          }
+          giveRank(rankArray,result);
+        });
+}
+
+function giveRank(arrayArg,resultArg){
+    let rank=1;
+    prev_rank=rank;
+    position=0;
+    console.log('\n-------OUR RESULTS------\n');
+    console.log('Name | Mark | Position\n');
+    for (i=0; i < arrayArg.length ; i++) {
+
+            if(i==0){
+                position=rank;
+            console.log (resultArg[i]['name']+"\t"+arrayArg[i]+"\t"+position)+"\n";
+            
+            
+            }else if(arrayArg[i]!=arrayArg[i-1]){
+            rank++;
+            position=rank;
+            prev_rank=rank;
+            console.log(resultArg[i]['name']+"\t"+arrayArg[i]+"\t"+position)+"\n";
+            
+            
+            }else{
+                position=prev_rank;
+                rank++;
+                console.log (resultArg[i]['name']+"\t"+arrayArg[i]+"\t"+position)+"\n";
+            }
+    }
+}
+```
```

Comment:
Instead of writing all the functions under one code snippet, and explaining at the end, you could break them down into several code snippets, and write its description along with it.

Reply:
And, don't forget to add comments.

Also, prettify the code using some extension. (Would enhance readability)